tests/CSharpTemp/CSharpTemp.h

```diff
@@ -175,8 +175,9 @@ class DLL_API P : Proprietor
 };
 
 // Tests destructors
-struct DLL_API TestDestructors
+class DLL_API TestDestructors                                  //Value Types can't have destructors
```

Comment:
I think you can use clang's `Decl::isImplicit()` to check if the declaration was explicitly declared by the user.